persistence-modules/hibernate-annotations/src/main/java/com/baeldung/hibernate/customtypes/SalaryType.java

```diff
@@ -73,27 +48,30 @@ public int hashCode(Salary x) {
 
     @Override
     public Salary nullSafeGet(ResultSet rs, int position, SharedSessionContractImplementor session, Object owner) throws SQLException {
-        Salary salary = new Salary();
-        salary.setAmount(rs.getLong(position));
-
+        
         if (rs.wasNull())
             return null;
+        
+        Salary salary = new Salary();
+        
+        String salaryValue = rs.getString(position);
+        
+        salary.setAmount(Long.parseLong(salaryValue.split(" ")[1]));
 
-        salary.setCurrency(rs.getString(position));
+        salary.setCurrency(salaryValue.split(" ")[1]);
```

Comment:
Maybe here we should get the first value, something like this: salaryValue.split(" ")[0], since the stored value is: USD 1000?